src/Sarif/Autogenerated/Result.cs

```diff
@@ -48,15 +48,51 @@ public SarifNodeKind SarifNodeKind
         [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
         public int RuleIndex { get; set; }
 
+        private ResultKind _kind;
+        /// <summary>
+        /// A value that categorizes results by evaluation state.
+        /// </summary>
+        [DataMember(Name = "kind", IsRequired = false, EmitDefaultValue = false)]
+        [DefaultValue(ResultKind.Fail)]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        [JsonConverter(typeof(Microsoft.CodeAnalysis.Sarif.Readers.EnumConverter))]
+        public ResultKind Kind
+        {
+            get { return _kind; }
+            set
+            {
+                _kind = value;
+                if (_kind != ResultKind.None && _kind != ResultKind.Fail)
```

Comment:
>_kind != ResultKind.None [](start = 20, length = 24)

I don't agree with the first part of the condition. If somebody writes `myResult.Kind = ResultKind.None` -- if they don't even know the _kind_ of the result -- it's fair to say they don't know the failure level, or even if it _is_ a failure. I'd just leave the second half of the condition.

Having said that, I'm uncomfortable with this side effect; I can't decide whether if violates the principle of least surprise.

Reply:
No, you're right. If ResultKind is not Fail, None is the only valid value for FailureLevel. Good catch.

---
In reply to: [252898602](https://github.com/Microsoft/sarif-sdk/pull/1249#discussion_r252898602) [](ancestors = 252898602)